plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -457,17 +461,53 @@ public static ColumnMapping timestampColumnMapping()
 
     public static ColumnMapping timestampColumnMapping(TimestampType timestampType)
     {
-        checkArgument(timestampType.getPrecision() <= MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
-        return ColumnMapping.longMapping(
+        if (timestampType.getPrecision() <= MAX_SHORT_PRECISION) {
+            return ColumnMapping.longMapping(
+                    timestampType,
+                    timestampReadFunction(timestampType),
+                    timestampWriteFunction(timestampType));
+        }
+        return ColumnMapping.objectMapping(
                 timestampType,
-                timestampReadFunction(timestampType),
-                timestampWriteFunction(timestampType));
+                longTimestampReadFunction(timestampType),
+                longTimestampWriteFunction(timestampType));
     }
 
     public static LongReadFunction timestampReadFunction(TimestampType timestampType)
     {
         checkArgument(timestampType.getPrecision() <= MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
-        return (resultSet, columnIndex) -> toPrestoTimestamp(timestampType, resultSet.getObject(columnIndex, LocalDateTime.class));
+        return (resultSet, columnIndex) -> {
+            long epochMicros = toPrestoTimestamp(timestampType, resultSet.getObject(columnIndex, LocalDateTime.class));
+            verifyTimestamp(epochMicros, 0, timestampType.getPrecision());
+            return epochMicros;
+        };
+    }
+
+    private static ObjectReadFunction longTimestampReadFunction(TimestampType timestampType)
+    {
+        checkArgument(timestampType.getPrecision() > MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
+        return ObjectReadFunction.of(
+                LongTimestamp.class,
+                (resultSet, columnIndex) -> {
+                    LongTimestamp longTimestamp = toLongTimestamp(timestampType, resultSet.getObject(columnIndex, LocalDateTime.class));
+                    verifyTimestamp(longTimestamp, timestampType.getPrecision());
+                    return longTimestamp;
+                });
+    }
+
+    public static void verifyTimestamp(LongTimestamp timestamp, int precision)
```

Comment:
inline

Reply:
Done! But I also removed the tests, is that ok? I guess tests for the whole read and write functions would require more cases.